src/sql/workbench/contrib/notebook/browser/notebookEditor.ts

```diff
@@ -8,37 +8,156 @@ import { BaseEditor } from 'vs/workbench/browser/parts/editor/baseEditor';
+import { Command } from 'vs/editor/browser/editorExtensions';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
+import { KeyMod, KeyCode } from 'vs/base/common/keyCodes';
+import { KeybindingWeight } from 'vs/platform/keybinding/common/keybindingsRegistry';
+import { NOTEBOOK_COMMAND_SEARCH, NOTEBOOK_COMMAND_CLOSE_SEARCH, NotebookEditorVisibleContext } from 'sql/workbench/services/notebook/common/notebookContext';
+import { IDisposable, DisposableStore, dispose } from 'vs/base/common/lifecycle';
+import { IModelDecorationsChangeAccessor, IModelDeltaDecoration } from 'vs/editor/common/model';
+import { NotebookFindDecorations, NotebookRange } from 'sql/workbench/contrib/notebook/browser/cellViews/NotebookFindDecorations';
+import { TimeoutTimer } from 'vs/base/common/async';
+import { BaseTextEditor } from 'vs/workbench/browser/parts/editor/textEditor';
 
-export class NotebookEditor extends BaseEditor {
+export class NotebookEditor extends BaseEditor implements IFindNotebookController {
 
 	public static ID: string = 'workbench.editor.notebookEditor';
 	private _notebookContainer: HTMLElement;
+	private _currentDimensions: DOM.Dimension;
+	private _overlay: HTMLElement;
+	private _findState: FindReplaceState;
+	private _finder: FindWidget;
+	private _actionMap: { [x: string]: IEditorAction } = {};
+	private _onDidChangeConfiguration = new Emitter<IConfigurationChangedEvent>();
+	public onDidChangeConfiguration: Event<IConfigurationChangedEvent> = this._onDidChangeConfiguration.event;
+	private _notebookModel: INotebookModel;
+	private _findCountChangeListener: IDisposable;
+	private _currentMatch: NotebookRange;
+	private _previousMatch: NotebookRange;
+	private readonly _decorations: NotebookFindDecorations;
+	private readonly _toDispose = new DisposableStore();
+	private readonly _startSearchingTimer: TimeoutTimer;
 
 	constructor(
 		@ITelemetryService telemetryService: ITelemetryService,
 		@IThemeService themeService: IThemeService,
-		@IInstantiationService private instantiationService: IInstantiationService,
-		@IStorageService storageService: IStorageService
+		@IInstantiationService private _instantiationService: IInstantiationService,
+		@IStorageService storageService: IStorageService,
+		@IContextViewService private _contextViewService: IContextViewService,
+		@IKeybindingService private _keybindingService: IKeybindingService,
+		@IContextKeyService private _contextKeyService: IContextKeyService,
+		@IWorkbenchThemeService private _themeService: IWorkbenchThemeService,
+		@INotebookService private _notebookService?: INotebookService
 	) {
 		super(NotebookEditor.ID, telemetryService, themeService, storageService);
+		this._startSearchingTimer = new TimeoutTimer();
+		this._decorations = new NotebookFindDecorations(this);
+		this._toDispose.add(this._decorations);
+		this._decorations.setStartPosition(this.getPosition());
```

Comment:
Is there a reason you're not just having this extend Disposable and use _register?

Reply:
well to be honest, this code was added by Lucy and I haven't had a second look at it.